src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -2865,6 +2978,23 @@ public bool Contains(TKey key)
                 } while (g != _lastGrouping);
             }
         }
+        
+        public IGrouping<TKey, TElement>[] ToArray()
```

Comment:
The embarrassing bit is I had this as it should be at the start of my coding it and for some mysterious reason changed it to be public.

> (If we want to make it public, which seems reasonable, it would need to go through the new API process).

It would, but it wouldn't be reasonable, IMO: Any time this class is available, `Enumerable.ToArray()` is available, and now just calls into this anyway, so the upside would be tiny. (A discoverable `Count` and `CopyTo` I would argue for, though I understand the counter-arguments, but not fully public there either).